src/components/application_manager/src/application_manager_impl.cc

```diff
@@ -1530,15 +1535,47 @@ void ApplicationManagerImpl::OnServiceEndedCallback(
   }
 }
 
+void AddAppIDForRPCService(bool& is_first_rpc_service_accepted,
+                           hmi_apis::Common_ServiceEvent::eType service_event,
+                           smart_objects::SmartObjectSPtr notification,
+                           const uint32_t connection_key) {
+  if (!is_first_rpc_service_accepted &&
+      service_event == hmi_apis::Common_ServiceEvent::REQUEST_ACCEPTED) {
+    is_first_rpc_service_accepted = true;
+  } else if (is_first_rpc_service_accepted) {
+    (*notification)[strings::msg_params][strings::app_id] = connection_key;
+  }
+}
+
 void ApplicationManagerImpl::ProcessStatusUpdate(
+    const uint32_t connection_key,
     hmi_apis::Common_ServiceType::eType service_type,
     hmi_apis::Common_ServiceEvent::eType service_event,
     hmi_apis::Common_ServiceUpdateReason::eType service_update_reason) {
   LOG4CXX_AUTO_TRACE(logger_);
+
+  static bool is_first_rpc_service_accepted = false;
+
+  LOG4CXX_DEBUG(logger_,
+                "Processing status update with connection key: "
+                    << connection_key << " service type: " << service_type
+                    << " service_event " << service_event
+                    << "service_update_reason " << service_update_reason);
+
   auto notification =
       MessageHelper::CreateOnServiceStatusUpdateNotification(service_type);
   (*notification)[strings::msg_params][hmi_notification::service_event] =
       service_event;
+
+  if (service_type == hmi_apis::Common_ServiceType::RPC) {
+    AddAppIDForRPCService(is_first_rpc_service_accepted,
```

Comment:
@mked-luxoft is_first_rpc_service_accepted is output variable. No not see reason for it.